src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/Diagnostics.cs

```diff
@@ -74,19 +76,41 @@ internal partial class RazorCustomMessageTarget
             return null;
         }
 
-        var request = new VSInternalDocumentDiagnosticsParams
-        {
-            TextDocument = identifierFromOriginalRequest.WithUri(virtualDocument.Uri),
-        };
-
         var lspMethodName = VSInternalMethods.DocumentPullDiagnosticName;
         using var _ = _telemetryReporter.TrackLspRequest(lspMethodName, delegatedLanguageServerName, correlationId);
-        var response = await _requestInvoker.ReinvokeRequestOnServerAsync<VSInternalDocumentDiagnosticsParams, VSInternalDiagnosticReport[]?>(
-            virtualDocument.Snapshot.TextBuffer,
-            lspMethodName,
-            delegatedLanguageServerName,
-            request,
-            cancellationToken).ConfigureAwait(false);
+
+        ReinvocationResponse<VSInternalDiagnosticReport[]?>? response = null;
+        if (_languageServerFeatureOptions.UseRoslynAnalyzerDiagnostics && delegatedLanguageServerName == RazorLSPConstants.RazorCSharpLanguageServerName)
+        {
+            var diagnosticParams = new RazorDiagnosticsParams
+            {
+                RazorTextDocument = identifierFromOriginalRequest,
+                TextDocument = identifierFromOriginalRequest.WithUri(virtualDocument.Uri),
```

Comment:
This won't work, I believe. `WithUri` modifies the original TextDocumentIdentifier, so both of these properties would end up pointing to the same Uri. 

Having said that, a `new TextDocumentIdentifier { Uri = virtualDocument.Uri }` should be sufficient any way, as the project context info that might be lost is only relevant to us, not Roslyn, and the virtual document Uris are unique per project anyway.

Reply:
I believe this to be fixed